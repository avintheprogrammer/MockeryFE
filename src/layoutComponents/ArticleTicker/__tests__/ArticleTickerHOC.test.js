import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { ArticleTicker } from '../index';
import { mock } from '../__mocks__/ArticleTickerHOC.mock';

configure({ adapter: new Adapter() });

test('Article Ticker renders correctly', () => {
  const tree = shallow(<ArticleTicker data={mock} />);
  expect(toJson(tree)).toMatchSnapshot();
});

test('Render the article header if data is valid.', () => {
  const wrapper = shallow(<ArticleTicker data={mock} />);
  expect(wrapper.text()).toBe('<Connect(Apollo(ArticleTicker)) />');
});
