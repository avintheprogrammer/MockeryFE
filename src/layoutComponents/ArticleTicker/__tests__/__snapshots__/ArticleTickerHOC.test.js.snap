// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Ticker renders correctly 1`] = `
<Connect(Apollo(ArticleTicker))
  cards={
    Array [
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Liberty Media Chairman John Malone shares his views in an exclusive interview with MOCKERY’s David Faber Thursday.",
        "id": "104846707",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104614395-GettyImages-520716340-jeff-bezos.jpg?v=1501169479",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "John Malone says Amazon is a 'Death Star' ",
        "tag": "",
        "title": "John Malone says Amazon is a 'Death Star' ",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/john-malone-says-amazon-is-a-death-star-moving-in-striking-range-of-every-industry-on-the-planet.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "An AT&T takeover of Time Warner, without a sale of CNN, is still quite likely, media mogul John Malone says. ",
        "id": "104846858",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104570724-GettyImages-628282192.jpg?v=1499350922",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "John Malone: AT&T-Time Warner deal still has 75% chance, with CNN included",
        "tag": "",
        "title": "John Malone: AT&T-Time Warner deal still has 75% chance, with CNN included",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/john-malone-att-time-warner-deal-still-has-75-percent-chance.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "President Trump's white working-class supporters don't gain much from the Republican tax cut push. They support it anyway.",
        "id": "104845427",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104143295-GettyImages-627104906.jpg?v=1481719301",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Trump’s core supporters are about to be handed the bill for tax reform",
        "tag": "",
        "title": "Trump’s core supporters are about to be handed the bill for tax reform",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/trumps-core-supporters-are-about-to-be-handed-the-bill-for-tax-reform.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Johnson says President Trump personally promised to \\"'work my tail off'\\" to win the Wisconsin Republican's vote.",
        "id": "104846845",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104653968-IMG_6501.JPG?v=1510778825",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Sen. Ron Johnson: Trump promises details so I can try to fix tax bill",
        "tag": "",
        "title": "Sen. Ron Johnson: Trump promises details so I can try to fix tax bill",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/sen-ron-johnson-trump-promises-details-so-i-can-try-to-fix-gop-tax-reform-bill.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Adding ACA repeal to tax reform is a distraction that could kill the bill and sink the market, says Jack Bouroudjian.",
        "id": "104847085",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104190690-GettyImages-182178417.jpg?v=1482991705",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "The market could be 'in the crosshairs’ of a 20% drop if tax reform fails",
        "tag": "",
        "title": "The market could be 'in the crosshairs’ of a 20% drop if tax reform fails",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/tax-reform-failure-could-cause-20-percent-market-drop-commentary.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Wal-Mart reports strong growth online, with e-commerce sales growing 50 percent in the fiscal third quarter.",
        "id": "104835075",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104166951-RTX2QKP4.jpg?v=1484848115",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Wal-Mart's earnings crush expectations as grocery and online sales surge",
        "tag": "",
        "title": "Wal-Mart's earnings crush expectations as grocery and online sales surge",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/wmt-q3-earnings-2017.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Secular bull Jeffrey Saut of Raymond James sees trouble for the rally.",
        "id": "104845029",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104183134-GettyImages-1300928.jpg?v=1490298268",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": " Wall Street bull warns a major pullback is near, and here’s what could spark it",
        "tag": "",
        "title": " Wall Street bull warns a major pullback is near, and here’s what could spark it",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/jeffrey-saut-major-stock-market-pullback-is-near-this-could-spark-it.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "The National Football League's weak viewership has Wall Street on edge, and the numbers aren't looking any better for Week 10.",
        "id": "104847087",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104847468-tropical_beach_204377.jpg?v=1516644087",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Wall Street's NFL worries mount as viewership slides with tough weeks ahead",
        "tag": "",
        "title": "Wall Street's NFL worries mount as viewership slides with tough weeks ahead",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/nfl-worries-climb-with-tough-weeks-ahead.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "These underwater bars, hotels and restaurants are sure to impress luxury travelers.",
        "id": "104819615",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104827242-Above-and-under-water.jpg?v=1510097004",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "See inside 5 of the world's most amazing underwater hotels and restaurants",
        "tag": "",
        "title": "See inside 5 of the world's most amazing underwater hotels and restaurants",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/07/underwater-hotels-and-restaurants-in-the-maldives-and-beyond.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Sales were boosted by the company's growing security business and subscription software. ",
        "id": "104846992",
        "img": "",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Cisco stock pops 6% after earnings beat improves turnaround outlook",
        "tag": "",
        "title": "Cisco stock pops 6% after earnings beat improves turnaround outlook",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/cisco-csco-stock-up-after-earnings-for-q1-2018.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Coca-Cola CEO James Quincey, who took over in May after serving as chief operating officer, reaffirmed guidance ahead of investor day.",
        "id": "104846991",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104846984-AP_623426446392.jpg?v=1515094220",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Coca-Cola CEO: 'Never say never' on getting into booze, but it 'doesn't make sense' ",
        "tag": "",
        "title": "Coca-Cola CEO: 'Never say never' on getting into booze, but it 'doesn't make sense' ",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/coca-cola-ceo-james-quincey.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Kairos Society launched a $25 million fund Thursday to back young entrepreneurs with big ideas to change the world and help society.",
        "id": "104842398",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/102183688-162492024.jpg?v=1492710500",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "A fund launches to disrupt Silicon Valley ",
        "tag": "",
        "title": "A fund launches to disrupt Silicon Valley ",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/fund-to-disrupt-silicon-valley-and-back-moonshots-for-social-good.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "\\"Jack's got more business sense than ever,\\" Jim Cramer says.",
        "id": "104846923",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104098135-IMG_3527.jpg?v=1478805921",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Cramer: Ex-CEO Jack Welch back on GE's board isn't a bad idea",
        "tag": "",
        "title": "Cramer: Ex-CEO Jack Welch back on GE's board isn't a bad idea",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/cramer-ex-ceo-jack-welch-back-on-ges-board-isnt-a-bad-idea.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Games from the Mountain West, West Coast Conference, Conference USA, and the Atlantic 10 divisions will be streamed live from the platform. ",
        "id": "104847005",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104847068-GettyImages-656637866.jpg?v=1510845510",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Facebook will stream exclusively 47 college basketball games this season",
        "tag": "",
        "title": "Facebook will stream exclusively 47 college basketball games this season",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/stadium-bring-47-college-basketball-games-exclusively-to-facebook.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Apple shares could trade between $210 and $230 over the next year, GBH Insights analyst Daniel Ives said in a note.
",
        "id": "104846687",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104818498-GettyImages-864097706-tim-cook.jpg?v=1509709181",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Apple could be a $1 trillion company within the next year, top analyst says ",
        "tag": "",
        "title": "Apple could be a $1 trillion company within the next year, top analyst says ",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/apple-1-trillion-company-within-the-next-year-analyst.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "Vail Resorts CEO Robert Katz was looking for ways to energize a boring business when he put a new twist on an old concept: the season pass.

",
        "id": "104844963",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/103703377-GettyImages-109820993.jpg?v=1510779089",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "How a resort CEO transformed his business with one simple idea",
        "tag": "",
        "title": "How a resort CEO transformed his business with one simple idea",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/how-vail-resorts-ceo-robert-katz-transformed-his-business.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "It feels like a private resort. ",
        "id": "104835774",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104835824-Gaga_Estate_POOL.jpg?v=1510346513",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Look inside the $10,000-a-night Airbnb mansion where Lady Gaga stayed ",
        "tag": "",
        "title": "Look inside the $10,000-a-night Airbnb mansion where Lady Gaga stayed ",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/13/10000-a-night-airbnb-mansion-where-lady-gaga-stayed-during-coachella.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "The Securities and Exchange Commission is best known as the no nonsense arm of U.S. financial law.",
        "id": "104844886",
        "img": "https://qa-aws02capi-img-origin.mockery.com/image/104844982-sec_fort_worth_twitter.png?v=1510775622",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Meet the man behind the SEC’s dad jokes on Twitter",
        "tag": "",
        "title": "Meet the man behind the SEC’s dad jokes on Twitter",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/2017/11/16/sec-fort-worth-twitter-jokes-meet-the-man-behind-the-account.html",
      },
      Object {
        "assetType": "mockerynewsstory",
        "brand": "",
        "description": "The economy faces higher deficits, reduced government benefits and near-zero interest rates unless Congress acts, Cleveland Fed President Loretta Mester said.",
        "id": "104846877",
        "img": "",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "Fed's Mester warns on rising deficits, calls for more immigration",
        "tag": "",
        "title": "Fed's Mester warns on rising deficits, calls for more immigration",
        "type": "",
        "url": "http://qa-aws01pub.mockery.com/2017/11/16/federal-reserve-president-says-country-needs-more-immigration-to-save-economy.html",
      },
      Object {
        "assetType": "mockeryvideo",
        "brand": "",
        "description": "",
        "id": "104846850",
        "img": "",
        "isLive": false,
        "sectionSubType": "",
        "shorterHeadline": "PRO Uncut: John Rogers Jr.",
        "tag": "",
        "title": "PRO Uncut: John Rogers Jr.",
        "type": "",
        "url": "http://qa-aws01web.mockery.com/video/2017/11/16/full-interview-with-john-rogers-jr.html",
      },
    ]
  }
  dataId={0}
/>
`;
